perf_test/PerfTest_Solvent.cpp

```diff
@@ -0,0 +1,261 @@
+  auto h_vexc = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), vexc);
+  auto h_w    = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), w);
+  auto h_delta_f_ref = Kokkos::create_mirror_view(delta_f_ref);
+  for (int i0 = 0; i0 < n0; i0++) {
+    for (int i1 = 0; i1 < n1; i1++) {
+      for (int i2 = 0; i2 < n2; i2++) {
+        for (int i3 = 0; i3 < n3; i3++) {
+          h_delta_f_ref(i0, i1, i2, i3) =
+              2.0 * rho0 * h_xi(i0, i1, i2, i3) * h_vexc(i0, i1, i2, i3);
+
+          ff_ref += rho0 * (h_xi(i0, i1, i2, i3) * h_xi(i0, i1, i2, i3) - 1.0) *
+                    prefactor * h_w(i3) * h_vexc(i0, i1, i2, i3);
+        }
+      }
+    }
+  }
+  Kokkos::deep_copy(delta_f_ref, h_delta_f_ref);
+
+  return std::make_tuple(xi, vexc, w, delta_f, delta_f_ref, ff_ref, rho0,
+                         prefactor);
+}
+
+template <typename T>
+auto test_get_delta_f(const View4DLayRType<T>& xi,
+                      const View4DLayRType<T>& vexc, const View1DLayRType<T>& w,
+                      const View4DLayRType<T>& delta_f,
+                      const View4DLayRType<T>& delta_f_ref, const T ff_ref,
+                      const T rho0, const T prefactor) {
+  execution_space exec_space;
+  T ff = 0;
+  exec_space.fence();
+  Kokkos::Timer timer;
+  MDFT::get_delta_f(exec_space, xi, vexc, w, delta_f, ff, rho0, prefactor);
+  exec_space.fence();
+  double time_delta_f = timer.seconds();
+
+  T epsilon   = std::numeric_limits<T>::epsilon() * 100;
+  bool passed = allclose(exec_space, delta_f, delta_f_ref, epsilon) &&
+                (Kokkos::abs(ff - ff_ref) < epsilon * Kokkos::abs(ff_ref));
+
+  return std::make_tuple(time_delta_f, passed);
+}
+
+BENCHMARK_TEMPLATE_DEFINE_F(TestSolvent, GetDeltaFFloat, float)
+(benchmark::State& state) {
+  int n0 = state.range(0), n1 = state.range(1), n2 = state.range(2),
+      n3 = state.range(3);
+
+  auto [xi, vexc, w, delta_f, delta_f_ref, ff_ref, rho0, prefactor] =
+      prep_delta_f<float>(n0, n1, n2, n3);
+  for (auto _ : state) {
+    const auto [time_delta_f, passed] = test_get_delta_f<float>(
+        xi, vexc, w, delta_f, delta_f_ref, ff_ref, rho0, prefactor);
+
+    state.SetIterationTime(time_delta_f);
+    state.counters["Passed"] = passed;
+  }
+}
+
+BENCHMARK_TEMPLATE_DEFINE_F(TestSolvent, GetDeltaFDouble, double)
+(benchmark::State& state) {
+  int n0 = state.range(0), n1 = state.range(1), n2 = state.range(2),
+      n3 = state.range(3);
+
+  auto [xi, vexc, w, delta_f, delta_f_ref, ff_ref, rho0, prefactor] =
+      prep_delta_f<double>(n0, n1, n2, n3);
+  for (auto _ : state) {
+    const auto [time_delta_f, passed] = test_get_delta_f<double>(
+        xi, vexc, w, delta_f, delta_f_ref, ff_ref, rho0, prefactor);
+
+    state.SetIterationTime(time_delta_f);
+    state.counters["Passed"] = passed;
+  }
+}
+
+static constexpr double WARMUP_TIME = 0.1;
+
+BENCHMARK_REGISTER_F(TestSolvent, GetDeltaRhoFloat)
```

Comment:
Can you make this a cartesian product. 
For the first three values `(nx, ny, nz)`, you can just use `n` to represent the number of spatial grid points. Scanning n from `30` to `100` should be good. 
For the 4th dim, it spans from `20` to `20`.